maintenance/cleanup.py

```diff
@@ -0,0 +1,182 @@
+    ctx = {'omero.group': '-1'}
+    images = query_service.findAllByQuery(query, params, ctx)
+    if len(images) == 0:
+        print "No image with name %s found" % image
+        return
+    params = omero.sys.ParametersI()
+    params.addLong('imageId', images[0].getId().getValue())
+    query = "from DatasetImageLink where child.id=:imageId"
+    links = conn.getQueryService().findAllByQuery(query, params, ctx)
+    if len(links) > 0:
+        ids = []
+        for link in links:
+            ids.append(link.getId().getValue())
+        conn.deleteObjects("DatasetImageLink", ids, deleteAnns=True,
+                           deleteChildren=False, wait=True)
+
+
+def list_objs(conn, username, target):
+    if target is None:
+        print "No dataset to handle"
+        return
+    params = omero.sys.ParametersI()
+    params.addString('username', username)
+    query = "from Dataset where name='%s' \
+             AND details.owner.omeName=:username" % target
+    query_service = conn.getQueryService()
+    ctx = {'omero.group': '-1'}
+    datasets = query_service.findAllByQuery(query, params, ctx)
+    if len(datasets) == 0:
+        print "No dataset with name %s found" % target
+        return
+
+    dataset_id = datasets[0].getId().getValue()
+    dataset = conn.getObject("Dataset", dataset_id)
+    roi_service = conn.getRoiService()
+    for image in dataset.listChildren():
+        result = roi_service.findByImage(image.getId(), None)
+
+        for roi in result.rois:
+            roi_id = roi.getId().getValue()
+            obj_ids_rois.append(roi_id)
+
+        if not len(result.rois) == 0:
+            value = 'Will delete %s ROIs on image \
+                    %s of %s' % (len(result.rois), image.getId(), username)
+            print value
+
+        for ann in image.listAnnotations():
+            if ann.OMERO_TYPE == omero.model.LongAnnotationI:
+                obj_ids_rating.append(ann.getId())
+                value = 'Will delete rating %s on image \
+                        %s of %s' % (ann.getId(), image.getId(), username)
+                print value
+            elif ann.OMERO_TYPE == omero.model.TagAnnotationI:
+                params = omero.sys.ParametersI()
+                params.add('imageId', wrap(image.getId()))
+                params.addId(ann.getId())
+                query = "select l.id from ImageAnnotationLink as \
+                        l where l.parent.id=:imageId AND l.child.id=:id"
+                linkIds = query_service.projection(query, params, ctx)
+                for linkId in linkIds:
+                    obj_ids_taglinks.append(linkId[0].getValue())
+                    value = 'Will delete link %s on image \
+                            %s of tag %s of %s' % (linkId[0].getValue(),
+                                                   image.getId(),
+                                                   ann.getId(), username)
+                    print value
+
+
+def delete_objs(conn):
```

Comment:
```
PYTHONPATH=$PYTHONPATH:. python ~/Work/training-scripts/maintenance/cleanup.py --name user-6 --dataset testing-name --server eel.openmicroscopy.org $PWD
No dataset with name testing-name found
No image to handle
user-1
No dataset with name testing-name found
No image to handle
user-2
No dataset with name testing-name found
No image to handle
user-3
No dataset with name testing-name found
No image to handle
user-4
Error while cleaning the objects: 'NoneType' object has no attribute 'listChildren'
```

Again, user-4 has the dataset, and I understand that it was found successfully. But probably the objects were not fully loaded when it came to this cleanup or something.... see above

Reply:
This should now be fixed